usr/share/rear/finalize/Linux-i386/670_run_efibootmgr.sh

```diff
@@ -19,60 +19,60 @@ is_true $EFI_STUB && return
-# if we have 'mapper' in disk device name:
-if [[ ${Dev/mapper//} != $Dev ]] ; then
-    # we only expect mpath_partX or mpathpX or mpath-partX
-    case $Disk in
-        (*p)     Disk=${Disk%p} ;;
-        (*-part) Disk=${Disk%-part} ;;
-        (*_part) Disk=${Disk%_part} ;;
-        (*)      Log "Unsupported kpartx partition delimiter for $Dev"
-    esac
-fi
+# Mount point inside the target system,
+# accounting for possible trailing slashes in TARGET_FS_ROOT
+esp_mountpoint_inside="${esp_mountpoint#${TARGET_FS_ROOT%%*(/)}}"
 
-# For eMMC devices the trailing 'p' in the Disk value
-# (as in /dev/mmcblk0p that is derived from /dev/mmcblk0p1)
-# needs to be stripped (to get /dev/mmcblk0), otherwise the
-# efibootmgr call fails because of a wrong disk device name.
-# See also https://github.com/rear/rear/issues/2103
-if [[ $Disk = *'/mmcblk'+([0-9])p ]] ; then
-    Disk=${Disk%p}
-fi
-
-# For NVMe devices the trailing 'p' in the Disk value
-# (as in /dev/nvme0n1p that is derived from /dev/nvme0n1p1)
-# needs to be stripped (to get /dev/nvme0n1), otherwise the
-# efibootmgr call fails because of a wrong disk device name.
-# See also https://github.com/rear/rear/issues/1564
-if [[ $Disk = *'/nvme'+([0-9])n+([0-9])p ]] ; then
-    Disk=${Disk%p}
+boot_efi_parts=$( find_partition "fs:$esp_mountpoint_inside" fs )
+if ! test "$boot_efi_parts" ; then
+    Log "Unable to find ESP $esp_mountpoint_inside in layout"
+    Log "Trying to determine device currently mounted at $esp_mountpoint as fallback"
+    boot_efi_dev="$( mount | grep "$esp_mountpoint" | awk '{print $1}' )"
+    if ! test "$boot_efi_dev" ; then
+        LogPrintError "Cannot create EFI Boot Manager entry (unable to find ESP $esp_mountpoint among mounted devices)"
+        return 1
+    fi
+    if test $(get_component_type "$boot_efi_dev") = part ; then
+        boot_efi_parts="$boot_efi_dev"
+    else
+        boot_efi_parts=$( find_partition "$boot_efi_dev" )
+    fi
+    if ! test "$boot_efi_parts" ; then
+        LogPrintError "Cannot create EFI Boot Manager entry (unable to find partition for $boot_efi_dev)"
+        return 1
+    fi
 fi
 
 # EFI\fedora\shim.efi
 BootLoader=$( echo $UEFI_BOOTLOADER | cut -d"/" -f4- | sed -e 's;/;\\;g' )
-LogPrint "Creating  EFI Boot Manager entry '$OS_VENDOR $OS_VERSION' for '$BootLoader' (UEFI_BOOTLOADER='$UEFI_BOOTLOADER')"
-Log efibootmgr --create --gpt --disk ${Disk} --part ${ParNr} --write-signature --label \"${OS_VENDOR} ${OS_VERSION}\" --loader \"\\${BootLoader}\"
-if efibootmgr --create --gpt --disk ${Disk} --part ${ParNr} --write-signature --label "${OS_VENDOR} ${OS_VERSION}" --loader "\\${BootLoader}" ; then
-    # ok, boot loader has been set-up - tell rear we are done using following var.
-    NOBOOTLOADER=''
-    return
-fi
 
+for efipart in $boot_efi_parts ; do
+    # /dev/sda1 or /dev/mapper/vol34_part2 or /dev/mapper/mpath99p4
+    Dev=$( get_device_name $efipart )
+    # 1 (must anyway be a low nr <9)
```

Comment:
@pcahyna 
do you understand what that old comment
```
# 1 (must anyway be a low nr <9)
```
actually means?
I understand nothing at all what it means.

Reply:
@jsmeix 
> do you understand what that old comment
> 
> ```
> # 1 (must anyway be a low nr <9)
> ```
> 
> actually means?

No. it caught my attention as well, but I was afraid to remove it.

Reply:
LOL! :-))
I think I also have had a look at that comment some time ago.
I will dig a bit to find out if there is a meaning...
If not let's simply remove what nobody understands.

Reply:
I found where I had a look at that comment some time ago
https://github.com/rear/rear/commit/e3ca4ce8987dcd475696f32ce49ea3b0bc3dbb2a
and where that comment was kept but the matching code was changed
https://github.com/rear/rear/commit/bf6742bec5181ad7987a4a705668cc1a3834bef4
```
- ParNr=$( echo ${Dev} | sed -e 's/.*\([0-9]\).*/\1/' )  # 1 (must anyway be a low nr <9)
...
+ ParNr=$( get_partition_number $Dev )  # 1 (must anyway be a low nr <9)
```
and finally where that comment originated and where it made sense
https://github.com/rear/rear/commit/70490d16a62ee92a7129c5300a23eb29c22c2ef7
```
+ ParNr=$( echo ${Dev} | sed -e 's/.*\([0-9]\).*/\1/' )  # 1 (must anyway be a low nr <9)
```

So that comment has become obsolete since
https://github.com/rear/rear/commit/bf6742bec5181ad7987a4a705668cc1a3834bef4
and that comment should now be removed to avoid further confusion what it means.

Reply:
Thanks for the investigation. I removed the comment now.